controllers/argocd/secret.go

```diff
@@ -313,9 +313,10 @@ func (r *ReconcileArgoCD) reconcileExistingArgoSecret(cr *argoproj.ArgoCD, secre
 		secret.Data[common.ArgoCDKeyServerSecretKey] = sessionKey
 	}
 
+	// reset the value to default only when secret.data field is nil
 	if hasArgoAdminPasswordChanged(secret, clusterSecret) {
 		pwBytes, ok := clusterSecret.Data[common.ArgoCDKeyAdminPassword]
-		if ok {
+		if ok && secret.Data[common.ArgoCDKeyAdminPassword] == nil {
```

Comment:
based on @svghadi's comment I think I would agree that instead of just doing this when the field is nil, we can get rid of this whole check and overwrite block so that this field is exclusively managed by the argocd-server

Reply:
@jaideepr97 - On second thought, should we continue with `cluster-secret` for password reset in case of nil for now, just to avoid any potential breaking change and get rid of this secret completely in operator redesign?

Reply:
@svghadi sounds reasonable 

@anandrkskd please run a quick test to see if the initial secret is created by argo-cd when running the operator. Then we can accordingly decide if we can skip the cluster secret creation in the redesign

Reply:
Installed the latest argocd operator, and created an instance
These were the secrets that were created, example-argocd-cluster contains the `admin.password` and `argocd-secret` contains the hashed value of the admin.password.
`argocd-initial-admin-secret` is not created by the operator

Adding the output of secrets created by argocd instance below.
```
kubectl get secrets -n argocd
NAME                                    TYPE                DATA   AGE
example-argocd-ca                       kubernetes.io/tls   3      28s
example-argocd-cluster                  Opaque              1      28s
example-argocd-tls                      kubernetes.io/tls   2      28s
example-argocd-default-cluster-config   Opaque              4      28s
argocd-secret                           Opaque              5      28s
```

Reply:
If we remove the part of creation of cluster secret and places where it updates the cluster-secret from the code, we see argo-cd creates `argocd-initial-admin-secret` and can be used to log-in using GUI or CLI, and password reset also works as expected.

[checkout code-changes made](https://github.com/anandrkskd/argocd-operator/pull/1/files)
```
$ kubectl get secret -n argocd              
NAME                                    TYPE                DATA   AGE
example-argocd-ca                       kubernetes.io/tls   3      5m33s
example-argocd-tls                      kubernetes.io/tls   2      5m33s
example-argocd-default-cluster-config   Opaque              4      5m33s
argocd-initial-admin-secret             Opaque              1      34m
argocd-secret                           Opaque              5      5m33s
                                                                                                                                                                                                                                                                                                                                          
$ kubectl get secret -n argocd argocd-initial-admin-secret -o yaml
apiVersion: v1
data:
  password: c1gxVzF0T0lDODNuRU1YYQ==
kind: Secret
metadata:
  creationTimestamp: "2024-02-22T06:21:35Z"
  name: argocd-initial-admin-secret
  namespace: argocd
  resourceVersion: "1167"
  uid: c170935e-0ab5-412c-ae55-055e93d9230f
type: Opaque

$ echo c1gxVzF0T0lDODNuRU1YYQ== | base64 -d                    
sX1W1tOIC83nEMXa%                                                                                                                                                                                                                                                                                                                                                                                                                  

$ argocd login --username admin --password sX1W1tOIC83nEMXa 127.0.0.1:8443                 
WARNING: server certificate had error: x509: “example-argocd-tls” certificate is not trusted. Proceed insecurely (y/n)? y
'admin:login' logged in successfully
Context '127.0.0.1:8443' updated
```

Reply:
Awesome. In that case, I am okay to merge this PR and get rid of operator managed secret completely in refactored code.